app/database/group_attempt_store_compute_all_group_attempts.go

```diff
@@ -160,24 +149,62 @@ func (s *GroupAttemptStore) ComputeAllGroupAttempts() (err error) {
 							ELSE NULL
 							END),
 						target_groups_attempts.score = IF(children_stats.id IS NOT NULL,
-							children_stats.average_score, target_groups_attempts.score),
-						target_groups_attempts.ancestors_computation_state = 'done'
+							children_stats.average_score, target_groups_attempts.score)
 					WHERE target_groups_attempts.ancestors_computation_state = 'processing'`
 				updateStatement, err = ds.db.CommonDB().Prepare(updateQuery)
 				mustNotBeError(err)
 				defer func() { mustNotBeError(updateStatement.Close()) }()
 			}
 
+			_, err = updateStatement.Exec()
+			mustNotBeError(err)
+
+			if unlockStatement == nil {
+				const unlockQuery = `
+					INSERT INTO permissions_granted
+						(group_id, item_id, giver_group_id, can_view, latest_update_on)
+						SELECT
+							groups.id AS group_id,
+							item_unlocking_rules.unlocked_item_id AS item_id,
+							-1,
+							'content',
+							NOW()
+						FROM groups_attempts
+						JOIN item_unlocking_rules ON item_unlocking_rules.unlocking_item_id = groups_attempts.item_id AND
+							item_unlocking_rules.score <= groups_attempts.score
+						JOIN ` + "`groups`" + ` ON groups_attempts.group_id = groups.id
+						WHERE groups_attempts.ancestors_computation_state = 'processing'
+					ON DUPLICATE KEY UPDATE
+						latest_update_on = IF(can_view = 'content', latest_update_on, NOW()),
+						can_view = GREATEST(can_view_value, 3 /* 'content' */)`
```

Comment:
As soon you'll have introduce the "origin" attribute, we know that it cannot be > than "can_view:content" so you can just ignore on duplicated. Maybe you should already do it this way directly.

Reply:
I'd prefer fixing it later together with introducing the 'origin' attribute.